includes/class-sensei-lesson.php

```diff
@@ -4061,12 +4140,13 @@ public static function course_signup_link() {
 		$show_course_signup_notice = sensei_is_login_required() && ! Sensei_Course::is_user_enrolled( $course_id );
 
 		/**
-		 * Filter for if we should show the course sign up notice on the lesson page.
+		 * Filter whether to show the course sign up notice on the lesson page.
 		 *
 		 * @since 2.0.0
+		 * @hook sensei_lesson_show_course_signup_notice
 		 *
-		 * @param bool $show_course_signup_notice True if we should show the signup notice to the user.
-		 * @param int  $course_id                 Post ID for the course.
+		 * @param {bool}         $show_course_signup_notice True if we should show the signup notice to the user.
+		 * @param {bool|string}  $course_id                 Course ID.
```

Comment:
This is either false or the course id as returned by `get_course_id`. It can't be false in the filter though since we return at line 4137. Same with the ones below.

Reply:
Fixed in https://github.com/Automattic/sensei/pull/3565/commits/d27c1a66a529917797628793f145955eaab1567a.